spec/jobs/metais/sync_document_job_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe Metais::SyncDocumentJob, type: :job do
+  describe '#perform' do
+    ActiveJob::Base.queue_adapter = :test
+    
+    let(:client) { instance_double(Faraday::Connection, get: faraday_response) }
+    let(:faraday_response) { instance_double(Faraday::Response, body: body) }
+    let(:body) { metais_json_fixture('document_response.json') }
+
+    let(:document) { build(:metais_project_document) }
+    let(:document_version) { build(:metais_project_document_version) }
+    let(:project) { build(:metais_project) }
+
+    before do
+      allow(Faraday).to receive(:new) { client } 
+      allow(client).to receive(:get) { faraday_response }
+      project.save
+    end
```

Comment:
tuto pripravu v klude do testu a ked potrebujes vytvorit project tak to daj do `let!` - s vykricnikom to priamo spusti lambda a dovnutra daj namiesto build create!